packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -413,6 +498,149 @@ export class Alarm extends AlarmBase {
   }
 }
 
+/**
+ * CloudWatch Alarm that uses anomaly detection to trigger alarms
+ *
+ * This alarm type is specifically designed for use with anomaly detection operators
+ * like LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD.
+ */
+export class AnomalyDetectionAlarm extends Alarm {
+  /**
+   * Creates an anomaly detection metric from the provided metric
+   *
+   * @param scope The construct scope
+   * @param props The anomaly detection alarm properties
+   * @returns An anomaly detection metric
+   */
+  static createAnomalyDetectionMetric(scope: Construct, props: AnomalyDetectionAlarmProps): IMetric {
+    // Validate stdDevs
+    if (props.stdDevs !== undefined && props.stdDevs <= 0) {
+      throw new ValidationError('stdDevs must be greater than 0', scope);
+    }
+
+    if (!Alarm.isAnomalyDetectionOperator(props.comparisonOperator)) {
+      throw new ValidationError(`Invalid comparison operator for anomaly detection alarm: ${props.comparisonOperator}`, scope);
+    }
+
+    // Create the anomaly detection band expression
+    const anomalyDetectionExpression = new MathExpression({
+      expression: `ANOMALY_DETECTION_BAND(m0, ${props.stdDevs ?? 2})`,
+      usingMetrics: { m0: props.metric },
+      period: props.period,
+      label: 'Anomaly Detection Band',
+    });
+
+    return anomalyDetectionExpression;
+  }
+
+  constructor(scope: Construct, id: string, props: AnomalyDetectionAlarmProps) {
+    super(scope, id, {
+      ...props,
+      metric: AnomalyDetectionAlarm.createAnomalyDetectionMetric(scope, props),
+      threshold: THRESHOLD_IGNORED_FOR_ANOMALY_DETECTION,
+    });
+  }
+
+  protected override onCfnAlarmCreated(alarm: CfnAlarm): void {
+    super.onCfnAlarmCreated(alarm);
+    this.ensureAnomalyMetricsReturnData(alarm);
```

Comment:
What's this protection against? Can we not set it up so it definitely works?

In any case, I don't like the pattern of making a new "subclass function hole" in this way.

Use `this.node.addValidation()` instead.